specification/resources/resource-manager/Microsoft.Resources/preview/2022-11-01-preview/snapshots.json

```diff
@@ -0,0 +1,247 @@
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/SnapshotListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Resources/snapshots/{snapshotId}": {
+      "get": {
+        "tags": [
+          "GetSnapshot"
+        ],
+        "operationId": "Snapshots_Get",
+        "x-ms-examples": {
+          "GetSnapshot": {
+            "$ref": "./examples/GetSnapshot.json"
+          }
+        },
+        "description": "Obtains the specified snapshot for the target resource",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceProviderNamespaceParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceTypeParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/snapshotIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/Snapshot"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Snapshot": {
+      "type": "object",
+      "description": "A generic resource snapshot with additional metadata",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SnapshotProperties"
+        }
+      }
+    },
+    "SnapshotProperties": {
+      "description": "The properties of a snapshot",
+      "type": "object",
+      "properties": {
+        "apiVersion": {
```

Comment:
>apiVersion [](http://example.com/codeflow?start=9&length=10)

I'm not sure if its useful to add this to your mode but I  assume its meant to be treated as as 'readOnly' 'true', if snapshots are never updated in place?

Reply:
I don't think it's super necessary to add `readOnly` `true` here since we don't offer any way for users to update these snapshot objects (no PATCH or PUTs, just GETs)